src/components/Convert/getPegStabilityQuote.ts

```diff
@@ -0,0 +1,46 @@
+import { BigNumber, ethers, providers, Signer } from "ethers";
+import { HlpConfig } from "../..";
+import { PSM_GAS_LIMIT } from "../../config/hlp";
+import { HPSM__factory } from "../../contracts/factories/HPSM__factory";
+import { Network } from "../../types/network";
+import { Quote } from "./Convert";
+
+let transactionFeeCache: Record<Network, BigNumber | null> = {
+  arbitrum: null,
+  ethereum: null,
+  polygon: null
+};
+
+const TRANSACTION_FEE_UNIT = ethers.utils.parseEther("1");
+
+export const getPegStabilityQuote = async ({
+  network,
+  fromAmount,
+  provider
+}: {
+  fromToken: string;
+  toToken: string;
+  network: Network;
+  fromAmount: BigNumber;
+  provider: providers.Provider | Signer;
+}): Promise<{ quote: Quote; feeBasisPoints?: BigNumber }> => {
+  const hpsmAddress = HlpConfig.HLP_CONTRACTS[network]?.HPSM;
+  if (!hpsmAddress) {
+    throw new Error("No HPSM for network '" + network + "'");
```

Comment:
Probably better to do template / backtickets
```suggestion
    throw new Error(`No HPSM for network ${network}`);
```